File: HW11/hw11.py
from collections import deque

class pySol():
    def floodFill(self, image, sr, sc, color):
        oldColor = image[sr][sc]
        if oldColor is color:
            return image
        
        w = len(image)
        l = len(image[0])
        queue = deque([(sr,sc)])
        
        while queue:
            r,c = queue.popleft()
            if image[r][c] is oldColor:
                image[r][c] = color
                if(r>0):
                    queue.append((r-1,c))
                if(r<w-1):
                    queue.append((r+1,c))
                if(c>0):
                    queue.append((r,c-1))
                if(c<l-1):
                    queue.append((r,c+1))
        
        return image
    
# sol = pySol()
# image = [[1, 1, 1],[1, 1, 0],[1, 0, 1]]
# for row in image:
#     print(row)
# print("\n")
# sol.floodFill(image,1,1,2)
# for row in image:
#     print(row)

        

File: HW11/hw11.java
import java.util.*;

public class hw11 {
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        int oldColor = image[sr][sc];
        if (oldColor == newColor) // prevents infinite loop
            return image;
        
        int w = image.length;
        int l = image[0].length;
        LinkedList<int[]> queue = new LinkedList<>();
        queue.add(new int[]{sr,sc});

        while(!queue.isEmpty()){
            int[] pix = queue.poll();
            int r = pix[0];
            int c = pix[1];

            if(image[r][c] == oldColor){
                image[r][c] = newColor;

                if(r>0)
                    queue.add(new int[]{r-1,c});
                if(r<w-1)
                    queue.add(new int[]{r+1,c});
                if(c>0)
                    queue.add(new int[]{r,c-1});
                if(c<l-1)
                    queue.add(new int[]{r,c+1});
            }
        }

        return image;

    }

    public static void main(String[] args){
        // int[][] image = {{1,1,1},{1,1,0},{1,0,1}};
        // hw11 test = new hw11();
        // for(int[]row:image)
        //     System.out.println(Arrays.toString(row));

        // int[][] result = test.floodFill(image,1,1,2);
        // System.out.println();
        // for(int[]row:result)
        //     System.out.println(Arrays.toString(row));
        // System.out.println();

    }
}
